src/Runtime/Distributions/Automata/Automaton.Determinization.cs

```diff
@@ -415,6 +422,37 @@ public bool Equals(WeightedStateSet that)
                     return true;
                 }
 
+                /// <summary>
+                /// Checks whether this object is equal to a given one. Weights are compared with
+                /// given precision.
+                /// </summary>
+                /// <param name="that">The object to compare this object with.</param>
+                /// <param name="precision">Maximum difference of weights for them to be considered equal</param>
+                /// <returns>
+                /// <see langword="true"/> if the objects are equal,
+                /// <see langword="false"/> otherwise.
+                /// </returns>
+                public bool EqualsWithPrecision(WeightedStateSet that, double precision)
+                {
+                    if (this.Count != that.Count)
+                    {
+                        return false;
+                    }
+
+                    for (var i = 0; i < this.Count; ++i)
+                    {
+                        var state1 = this[i];
+                        var state2 = that[i];
+                        if (state1.Index != state2.Index ||
+                            Math.Abs(state1.Weight.LogValue - state2.Weight.LogValue) > precision)
```

Comment:
Choice between the absolute and the relative difference needs a justification. Intuitively, the log-values are expected to have highly varying magnitudes, so relative difference would seem more appropriate. The fact that this function is being used to check, if the difference between weights is explainable by a bunch of round-off errors points in the same direction.

Also, consider using `MMath.AbsDiff` (btw, this name is confusing - one of the overloads actually computes the relative difference).

Reply:
Probabilities are normalized at this point.
I.e. it is guaranteed that the state with highest probability has LogValue = 0.
So precision is effectively relative.

I will switch to relative diff anyway since it costs us nothing to do so.